bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/factory/KNXHandlerFactory.java

```diff
@@ -58,14 +62,16 @@ public class KNXHandlerFactory extends BaseThingHandlerFactory {
     @Nullable
     private final NetworkAddressService networkAddressService;
     private final SerialPortManager serialPortManager;
+    private final ThingRegistry thingRegistry;
```

Comment:
```suggestion
    private final Map<ThingUID, KNXBaseBridgeHandler> bridges = new ConcurrentHashMap<>();
```

I can't add suggestions to non-changed parts: what's missing is

```
        if (THING_TYPE_IP_BRIDGE.equals(thingTypeUID)) {
            ThingUID ipBridgeUID = getIPBridgeThingUID(thingTypeUID, thingUID, configuration);
            KNXBaseBridgeHandler bridgeHandler = super.createThing(thingTypeUID, configuration, ipBridgeUID, null);
            bridges.put(thing.getUID(), bridgeHandler);  
return bridgeHandler;
        }
```

and the same for serial bridges. Then add

```
    @Override
    public void unregisterHandler(Thing thing) {
        bridges.remove(thing.getUID();
        super.unregisterHandler(thing);
    }
```
it does not hurt to call it for every thing, nothing will be removed if the thing UID is not present.

Reply:
thanks for the input, done in a way similar to the suggested change....